Automattic/wp-calypso
# Add new paid stats site features: STATS_FREE, STATS_COMMERCIAL

Related to https://github.com/Automattic/red-team/issues/297

## Proposed Changes

* Adds new paid stats site features: STATS_FREE, STATS_COMMERCIAL
* Updates shouldGateStats to make use of the new site features

TODO:
- [x] We still need to add something to handle 2024-09-01 - 2024-12-tbd free sites, probably STATS_BASIC p1733289129342149/1733176661.084479-slack-C068X5XJT3J

## Testing Instructions

Test with / without the flag enabled:

```
?flags=stats/paid-wpcom-v3 (with, default)
?flags=-stats/paid-wpcom-v3 (without)
```

Test with / without the diff applied to your sandbox:

D167647-code

Test sites:
  - Free before 2024-01-09 (Should see v2 gating on UTM/Devices only - shouldn't actually see these at all (yet))
  - Free after 2024-01-09 (Should see v2 gating)
  - Free after 2024-12-tbd (Should see v3 gating)
  - Personal before 2024-01-09 (Should see v2 gating on UTM/Devices only)
  - Personal after 2024-01-09 (Should see less v2 gating - clicks+referrers should not be gated)
  - Premium before and after 2024-01-09 (Should see no gating)

Remember to test Jetpack/Atomic sites too, will need https://github.com/Automattic/jetpack/pull/40421

Unit tests:
```
yarn run test-client client/my-sites/stats/hooks/test/use-should-gate-stats.ts
```
(Should probably improve these now, I've only made them work with the new stats break down)

E2E:
```
yarn workspace wp-e2e-tests decrypt-secrets (only if you need to)
yarn workspace wp-e2e-tests test -- test/e2e/specs/stats/stats.ts
```
